com.sap.cloud.lm.sl.cf.core/src/main/java/com/sap/cloud/lm/sl/cf/core/parser/UriParametersParser.java

```diff
@@ -113,8 +116,26 @@ private void addPortBasedUris(Set<String> uris, String domain, List<Integer> por
     }
 
     public List<String> getApplicationRoutes(List<Map<String, Object>> parametersList) {
-        String routesParameterName = SupportedParameters.SINGULAR_PLURAL_MAPPING.get(routeParameterName);
-        return getAll(parametersList, routeParameterName, routesParameterName);
+        String route = routeParameterName != null ? (String) PropertiesUtil.getPropertyValue(parametersList, routeParameterName, null)
+            : null;
+
+        List<Map<String, Object>> routes = routesParameterName != null
+            ? (List<Map<String, Object>>) PropertiesUtil.getPropertyValue(parametersList, routesParameterName, Collections.emptyList())
+            : null;
+
+        List<String> allRoutesFound = new ArrayList<>();
+
+        if (route != null) {
+            allRoutesFound.add(route);
+        }
+        if (routes != null && !routes.isEmpty()) {
+            routes.stream()
```

Comment:
From 132-135, I believe there could be a better way to be done and all the ifs to be replaced with only one lambda. :)